src/plugins/console/public/lib/mappings/mappings.js

```diff
@@ -272,15 +281,15 @@ function retrieveSettings(settingsKey, settingsToRetrieve) {
 
   // Fetch autocomplete info if setting is set to true, and if user has made changes.
   if (settingsToRetrieve[settingsKey] === true) {
-    return opensearch.send('GET', settingKeyToPathMap[settingsKey], null);
+    return opensearch.send(http, 'GET', settingKeyToPathMap[settingsKey], null);
   } else {
-    const settingsPromise = new $.Deferred();
+    // const settingsPromise = new $.Deferred();
     if (settingsToRetrieve[settingsKey] === false) {
       // If the user doesn't want autocomplete suggestions, then clear any that exist
-      return settingsPromise.resolveWith(this, [[JSON.stringify({})]]);
+      return Promise.resolve({});
```

Comment:
It looks like the previous returned `'{}'` but now we return `{}`. Is this intentional?

Reply:
yes, the return type is now `Promise<HttpResponse>`, returning an object `{}` makes more sense than a string.